pkg/portal/cluster.go

```diff
@@ -190,3 +195,88 @@ func (p *portal) machineSets(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	_, _ = w.Write(b)
 }
+
+func (p *portal) statistics(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	duration := r.URL.Query().Get("duration")
+	endTimeString := r.URL.Query().Get("endtime")
+	endTime, err := time.Parse(time.RFC3339, endTimeString)
+	if err != nil {
+		p.internalServerError(w, err)
+		return
+	}
+	apiVars := mux.Vars(r)
+	subscription := apiVars["subscription"]
+	resourceGroup := apiVars["resourceGroup"]
+	clusterName := apiVars["clusterName"]
+	statisticsType := apiVars["statisticsType"]
+	promQuery := ""
+
+	switch statisticsType {
+	//kube-apiserver
+	case "kubeapicodes":
+		promQuery = "sum(rate(apiserver_request_total{job=\"apiserver\",code=~\"[45]..\"}[10m])) by (code, verb)"
+	case "kubeapicpu":
+		promQuery = "rate(process_cpu_seconds_total{job=\"apiserver\"}[5m])"
+	case "kubeapimemory":
+		promQuery = "process_resident_memory_bytes{job=\"apiserver\"}"
+	//kube-controller-manager
+	case "kubecontrollermanagercodes":
+		promQuery = "sum(rate(rest_client_requests_total{job=\"kube-controller-manager\"}[5m])) by (code)"
+	case "kubecontrollermanagercpu":
+		promQuery = "rate(process_cpu_seconds_total{job=\"kube-controller-manager\"}[5m])"
+	case "kubecontrollermanagermemory":
+		promQuery = "process_resident_memory_bytes{job=\"kube-controller-manager\"}"
+	//DNS
+	case "dnsresponsecodes":
+		promQuery = "sum(rate(coredns_dns_responses_total[5m])) by (rcode)"
+	case "dnserrorrate":
+		promQuery = "sum(rate(coredns_dns_responses_total{rcode=~\"SERVFAIL|NXDOMAIN\"}[5m])) by (pod) / sum(rate(coredns_dns_responses_total{rcode=~\"NOERROR\"}[5m])) by (pod)"
+
+	case "dnshealthcheck":
+		promQuery = "histogram_quantile(0.99, sum(rate(coredns_health_request_duration_seconds_bucket[5m])) by (le))"
+	case "dnsforwardedtraffic":
+		promQuery = ""
```

Comment:
is it empty on purpose?

Reply:
ohh no .. I need to put in a query there.

Reply:
I did put that query in the latest push, resolving.